cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -395,6 +396,35 @@ func (m *AwsManager) buildNodeFromTemplate(asg *asg, template *asgTemplate) (*ap
 	node.Status.Capacity[gpu.ResourceNvidiaGPU] = *resource.NewQuantity(template.InstanceType.GPU, resource.DecimalSI)
 	node.Status.Capacity[apiv1.ResourceMemory] = *resource.NewQuantity(template.InstanceType.MemoryMb*1024*1024, resource.DecimalSI)
 
+	if asg.MixedInstancesPolicy != nil {
+		instanceRequirements, err := m.getInstanceRequirementsFromMixedInstancesPolicy(asg.MixedInstancesPolicy)
+		if err != nil {
+			klog.Error("error while building node template using instance requirements: (%s)", err)
```

Comment:
in the case of an err returned in `getInstanceRequirementsFromMixedInstancesPolicy(...)` instanceRequirements will be nil. So I believe the next group of if's would panic on a nil pointer, right?

```
if instanceRequirements.VCpuCount != nil {
...
```

Reply:
You're right, thanks for catching. Will fix and make this section more readable as well.